action/protocol/poll/candidateindexer_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2019 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package poll
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/iotexproject/iotex-core/action/protocol/vote"
+	"github.com/iotexproject/iotex-core/db"
+	"github.com/iotexproject/iotex-core/state"
+	"github.com/iotexproject/iotex-core/test/identityset"
+)
+
+func TestCandidateIndexer(t *testing.T) {
+	require := require.New(t)
+	indexer, err := NewCandidateIndexer(db.NewMemKVStore())
+	require.NoError(err)
+	require.NoError(indexer.Start(context.Background()))
+	candidates := state.CandidateList{
+		{
+			Address:       identityset.Address(1).String(),
+			Votes:         big.NewInt(30),
+			RewardAddress: "rewardAddress1",
+		},
+		{
+			Address:       identityset.Address(2).String(),
+			Votes:         big.NewInt(22),
+			RewardAddress: "rewardAddress2",
+		},
+		{
+			Address:       identityset.Address(3).String(),
+			Votes:         big.NewInt(20),
+			RewardAddress: "rewardAddress3",
+		},
+		{
+			Address:       identityset.Address(4).String(),
+			Votes:         big.NewInt(10),
+			RewardAddress: "rewardAddress4",
+		},
+	}
+	require.NoError(indexer.PutCandidateList(uint64(1), &candidates))
```

Comment:
let's also try to call PutCandidateList twice with the same key and value, as well as with a new pair of key and value

Reply:
ok, but why do we have to test same key and value twice?

Reply:
in case that multiple writes may have different behavior. Actually, it is possible with current code base that it will be written twice with the same value.